source/features/sync-pr-commit-title.tsx

```diff
@@ -25,6 +25,10 @@ export function formatPrCommitTitle(title: string): string {
 	return `${title} (#${getConversationNumber()!})`;
 }
 
+export async function formatPrOrCompareCommitTitle(title: string): Promise<string> {
+	return `${title} (#${getConversationNumber() || await getNextConversationNumber()})`;
```

Comment:
This function is duplicating `formatPrCommitTitle` without actually replacing it. You might as well just create a dedicated `formatCompareCommitTitle` function with only `getNextConversationNumber` and move it out of `sync-pr-commit-title.tsx`

Reply:
yeah, I opted to duplicate it to make the pattern `${title} (#...)` as visually close together in the code as possible, to avoid the pattern implementations getting out of sync

if you don't think it's a risk, then happy to move this function out, eg. into `source/features/suggest-commit-title-limit.tsx` ... at that point we might even just inline the expression in the `validatePrTitle()` function instead of keeping it in a separate function 🤔

Reply:
an alternative would be that we could keep the single function and instead pass in the PR title function:

```ts
export function formatPrCommitTitle(
  title: string,
  getPrNumber = () => getConversationNumber()!
): string {
	return `${title} (#${getPrNumber()!})`;
}
```

then the call inside of `source/features/suggest-commit-title-limit.tsx` could "inject" this expression dependency:

```ts
async function validatePrTitle({delegateTarget: field}: DelegateEvent<Event, HTMLInputElement>): Promise<void> {
	// Include the PR number in the title length calculation because it will be added to the commit title
	const prTitle = await formatPrOrCompareCommitTitle(field.value, () => {
      return getConversationNumber() ?? await getNextConversationNumber()
    });
	field.classList.toggle('rgh-title-over-limit', prTitle.length > limit);
}
```

Reply:
Originally I considered just making `formatPrCommitTitle()` async, but this is used in quite a few places which do not need to call `getNextConversationNumber()`, which would add unnecessary `async/await` all over the place

Reply:
I'm ok with duplicating it, but it doesn't belong here anymore

Reply:
what about alternative 3 of passing in the PR number as an expression? feels cleaner somehow...

Reply:
Yeah that looks good too

Reply:
done in 675ab9511425633e3f6ec92890bf2b7bbf7cea6c